front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx

```diff
@@ -19,6 +28,74 @@ const Vias = ({ zoomToFeaturePoint }: ViasProps) => {
   const vias = useSelector(getVias());
   const pathSteps = useSelector(getPathSteps);
   const { launchPathfinding } = useManageTrainScheduleContext();
+  const infraId = useInfraID();
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+  const dispatch = useAppDispatch();
+
+  const hasUpdatedNames = useRef(false);
+
+  const searchOperationalPointsByUIC = useCallback(async () => {
+    if (!infraId || !pathSteps.length) return [];
+
+    const validPathSteps = pathSteps.filter(
+      (step): step is PathStep & { uic: number; secondary_code?: string } =>
```

Comment:
I would prefer something like 'step is Extract<PathStep, { uic: number }>' to assume as little as possible from what we actually check, but in this case I believe the cast is entirely unnecessary and typescript will do the correct assumptions by itself.
```suggestion
      (step) =>
```